File: php/features.php
<?php
declare(strict_types=1);

// Datamuse SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DatamuseFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DatamuseBaseFeature();
            case "test":
                return new DatamuseTestFeature();
            default:
                return new DatamuseBaseFeature();
        }
    }
}

File: py/features.py
# Datamuse SDK feature factory

from feature.base_feature import DatamuseBaseFeature
from feature.test_feature import DatamuseTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DatamuseBaseFeature(),
        "test": lambda: DatamuseTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
